sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DefaultDocumentQueryExecutionContext.java

```diff
@@ -186,7 +186,11 @@ protected PartitionKeyInternal getPartitionKeyInternal() {
                                                     this.fetchExecutionRangeAccumulator.getExecutionRanges(),
                                                     Arrays.asList(schedulingTimeSpanMap)),
                                             tFeedResponse.getActivityId());
-                            BridgeInternal.putQueryMetricsIntoMap(tFeedResponse, DEFAULT_PARTITION_RANGE, qm);
+                            String partitionKeyRange = tFeedResponse.getResponseHeaders().get(HttpConstants.HttpHeaders.PARTITION_KEY_RANGE_ID);
```

Comment:
@mbhaskar  please have a look and see if this proper way of fetching partition id

Reply:
why can't we rely on the partitionKeyRange instead? we are getting away from pkId.

@FabianMeiswinkel @mbhaskar thought?

Reply:
This is the part of diagnostics for query, and range is in not readable form , also hard for on call to check on kusto.

Reply:
@simplynaveen20 As discussed, we need to have the range info there. So lets have both range and id.

Reply:
Done